lib/bundle/mac_app_store_checker.rb

```diff
@@ -0,0 +1,39 @@
+# frozen_string_literal: true
+
+module Bundle
+  module MacAppStoreChecker
```

Comment:
These modules all look very similar. Would it be possible to DRY them up at all? I'm happy to give it a go myself if that'd be useful.

Reply:
It's very likely they can be DRYed up. BrewChecker and CaskChecker are virtually the same, with only arg names and the actionable package string as differences. MacAppStoreChecker's `exit_early_check` is also the same as the others, but its `full_check` is necessarily different because of the appid/name pair instead of a string.

```ruby
def full_check(app_ids)
  actionable = app_ids.reject { |id, _name| installed_and_up_to_date? id }
  actionable.map { |_id, name| "App #{name} needs to be installed or updated." }
end
```

If the checkers were classes instead of modules, the DRYed up version could have GenericChecker or something and inherit the functionality. I kept them as modules because that was the pattern established elsewhere throughout the code, IIRC (that decision was six+ months ago?)

Reply:
Yeh, I reckon classes instead of modules would make this a bit cleaner 👍

Reply:
Ideally part of this PR?